### PR TITLE
PARQUET-1917: Don't write values for oneOf fields that aren't set

### PR DESCRIPTION
**Pull request by dossett**
https://issues.apache.org/jira/browse/PARQUET-1917

### Jira

- [X] My PR addresses the following [Parquet Jira](https://issues.apache.org/jira/browse/PARQUET-1917/) issues and references them in the PR title. 

### Tests

- [X] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [X] My commits all reference Jira issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [X] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain Javadoc that explain what it does

### CODE REVIEW

File: parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoWriteSupport.java
Comment (dossett) [inline parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoWriteSupport.java:344]: Maybe this approach is too simplistic, but it does seem like exactly the behavior needed.

Comment (belugabehr) [reply parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoWriteSupport.java:344]: What about the read side?  Is there going to be any issue there?  Currently a value is always written, so there is a value to read.  Now that value is missing.

Comment (dossett) [reply parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoWriteSupport.java:344]: @belugabehr That's a great question.  Everything seems ok when I tested it out locally. I wrote some data out to a parquet file with this change, read the data back into protobuf messages, and then wrote the data back to a second parquet file.  The two parquet files were identical, so everything round tripped successfully.  

I am struggling a bit to write unit tests for this case, but that's probably me getting up the parquet API learning curve.

Comment (dossett) [reply parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoWriteSupport.java:344]: To be clear, I will try to add tests for this to my PR.

**Comment by belugabehr** [comment]
Thanks for the collaboration here.

For the unit tests you are using Java 'assert' functions.  It is required to use the JUnit Assert methods (look at other tests for reference).

**Comment by belugabehr** [comment]
Please add a test for setting field 1 and not setting field 2.  Since setting the fields should reset the values for all other fields in the 'oneof' field, it could be possible that field 2, set to '0', would clear the value of the first field.  If it just so happens that Parquet is loading the fields in the order they are defined, field 2 would be last and would overwrite all the previous values, so we don't really understand here what is happening to those fields.

**Comment by dossett** [comment]
Great feedback as always @belugabehr !

File: parquet-protobuf/src/test/java/org/apache/parquet/proto/ProtoWriteSupportTest.java
Comment (belugabehr) [inline parquet-protobuf/src/test/java/org/apache/parquet/proto/ProtoWriteSupportTest.java:964]: Bit of a nit here, but if you find yourself putting in comments like "this is test 1, this is test 2, this is test 3, etc." that is a clear sign that you need to break the test into multiple test methods.

If a test fails, it will be much easier to figure out exactly what broke instead of having to step through all this code.  It also ensure that the tests are testing what is intended and that there aren't any side-effects/artifacts of the other test scenarios interfering.

**Comment by dossett** [comment]
@gszadovszky Tagging you on this PR per discussion in the dev list.  If you approve the change I will also clean up the new tests a bit per David's comments.  Thanks!

**Comment by gszadovszky** [comment]
@dossett, it looks good to me. Let me wait for @belugabehr's approval then I'll approve and push this.

**Review by belugabehr — APPROVED** [review]

**Comment by belugabehr** [comment]
I looked at the unit test again, it's fine as-is.  +1

Thanks.

**Review by gszadovszky — APPROVED** [review]

**Comment by dossett** [comment]
Thank you both!

**Pull request merged**
